auction-app-frontend/src/components/FormContainer/FormContainer.jsx

```diff
@@ -1,32 +1,95 @@
 import { FormProvider } from "react-hook-form";
-import { InputField, Button } from "src/components";
+import { useState } from "react";
+
+import { InputField, TextAreaField, SelectField, CustomFileUploader, Button } from "src/components";
 import { BUTTON_VARIANTS } from "src/constants";
 
-const FormContainer = ({ formFields, onSubmit, buttonLabel, methods, error, buttonVariant = BUTTON_VARIANTS.FILLED, buttonIcon, children }) => {
+const FormContainer = ({
+  formFields,
+  onSubmit,
+  buttonLabel,
+  methods,
+  error,
+  buttonVariant = BUTTON_VARIANTS.FILLED,
+  buttonIcon,
+  children,
+  onCancel,
+  cancelLabel,
+  onBack,
+  backLabel
+}) => {
+  const [files, setFiles] = useState([]);
+
   return (
-    <FormProvider { ...methods }>
+    <FormProvider {...methods}>
+
       <form onSubmit={ methods.handleSubmit(onSubmit) }>
-        { children ? children : formFields.map(field => ( // children is used to render custom fields 
-          <InputField
-              key={ field.name }
-              name={ field.name }
-              label={ field.label }
-              type={ field.type }
-              rules={ field.rules }
-              step={ field.type === "number" ? field.step : null }
-              options={ field.type === "select" ? field.options : null }
-              className={ field.specialClass ? field.specialClass : "" }
-          />
-        )) }
-        { error && 
+        { children ? children : formFields.map((field) => {
+          switch (field.type) {
+            case "textarea":
```

Comment:
We can those types keep in constants, and read from there, just to follow current practices.